airbyte-integrations/connectors/source-faker/source_faker/streams.py

```diff
@@ -0,0 +1,275 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import datetime
+import os
+from typing import Any, Dict, Iterable, List, Mapping, Optional
+
+from airbyte_cdk.models import AirbyteEstimateTraceMessage, AirbyteTraceMessage, EstimateType, SyncMode, TraceType
+from airbyte_cdk.sources.streams import IncrementalMixin, Stream
+from mimesis import Datetime, Numeric, Person
+from mimesis.locales import Locale
+
+from .utils import format_airbyte_time, read_json
+
+
+class Products(Stream, IncrementalMixin):
+    primary_key = None
+    cursor_field = "id"
+
+    def __init__(self, count: int, seed: int, records_per_sync: int, records_per_slice: int, **kwargs):
+        super().__init__(**kwargs)
+        self.seed = seed
+        self.records_per_sync = records_per_sync
+        self.records_per_slice = records_per_slice
+
+    @property
+    def state_checkpoint_interval(self) -> Optional[int]:
+        return self.records_per_slice
+
+    @property
+    def state(self) -> Mapping[str, Any]:
+        if hasattr(self, "_state"):
+            return self._state
+        else:
+            return {self.cursor_field: 0}
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        self._state = value
+
+    def load_products(self) -> list[Dict]:
+        dirname = os.path.dirname(os.path.realpath(__file__))
+        return read_json(os.path.join(dirname, "record_data", "products.json"))
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
```

Comment:
python best practice: if you're not using the input params to a method, you can replace them with `**kwargs` to make it clear to the reader you don't care about it (also makes it easier to change method signatures, but that's less applicable here)